### PR TITLE
Add All and Between from IndexedTables.jl

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaComputing/IndexedTables.jl/issues/261

### CODE REVIEW

**Comment by bkamins** [comment]
@piever + @quinnj Do you have an opinion on this change? (after the change IndexedTables.jl should be updated and I will update DataFrames.jl)

**Comment by piever** [comment]
Sorry for the late reply! I'm in favor of this change and would like to take advantage of it to make the `All` constructor a bit stricter, i.e. leaving only one internal constructor `All(args...)` (same API as `union(args...)` from Base).

**Comment by bkamins** [comment]
@piever Is my change in line with what you intended?

**Comment by piever** [comment]
Yes, looks good to me!

**Comment by bkamins** [comment]
@quinnj or @nalimilan - can please either of you approve this before merging.

**Comment by quinnj** [comment]
This is fine this time, but consider adding tests next time; I realize it's mostly trivial, but it's at least good practice to ensure we don't have typos or anything.

**Comment by bkamins** [comment]
Sure. Actually I have looked at the tests and concluded the policy is not to add tests for bare definitions (but now - looking at them again - I see that you do, but simply there is not much to test yet). I will add tests in a separate PR. Thank you.

File: src/DataAPI.jl [hunk 1]
```diff
@@ -73,4 +73,22 @@ definition.
 """
 function describe end
 
+"""
+    Between(first, last)
+Select the columns between `first` and `last` from a table.
```
Comment (nalimilan) [inline src/DataAPI.jl]: Sorry to be late, but I think the convention is to have an empty row after the signature, right?

Comment (bkamins) [reply src/DataAPI.jl]: Right - I will add the fix (I copy pasted this from IndextedTables.jl, but now I see that it is correct there)

**Pull request merged**